_posts/2024-01-01-roadmap.md

```diff
@@ -0,0 +1,181 @@
+
+- **Sponsorship**: In 2023 we made a deliberate effort to "get over being shy" and ask for financial support, setting up a [sponsorship](https://geoserver.org/sponsor/) page, and listing sponsors on our [home page](https://geoserver.org/).
+
+  We received $1000 USD. You might think of this as a poor response.
+  
+  [North River Geographic Systems Inc](https://northrivergeographic.com/) provided funding to thank Andrea Aime for speaking at an event with no clear expectation of sponsorship. [How 2 Map](https://www.how2map.com/) sponsorship reflects Jody's personal company being used for screen snaps on how to badge a [github repository](https://github.com/jodygarnett/) as supporting OSGeo.
+  
+  With this in mind - no funds were directly raised in answer to our 2023 call for financial support. So this is actually a **terrible** response.
+  
+  *We ask for your financial assistance in 2024.*
+
+The above priorities of maintenance, testing and sponsorship represent the normal operations of an open-source project. This post is provided as a reminder, and a call to action for our community.
+
+### Roadmap Planning
+
+We have shared the following roadmap planning information in [foss4g presentations](https://docs.google.com/presentation/d/1X7iU1fd47frfh1EsN_CdUll0qtMMgPXkkMjaTbejj3g/edit?usp=sharing) in 2023, and it is time to share these goals with a wider audience as part of this blog post.
+
+![](/img/posts/2.25/roadmap_2024.svg)
+
+This is a brave step for the project: as we learned early on that placing a goal on a roadmap can be taken as an indication that funding is already secured. We even had a negative example where placing a goal on a roadmap resulted in the interested party withdrawing (as they understood that the community was now going to do the work instead!)
+
+With this in mind here are our priorities for 2024:
+
+* **Migrate to spring-framework-6** (Deadline December 2024)
+
+  GeoServer uses the spring-framework 5.3 which will reach end-of-life in December 2024. This provides motivation for all roadmap planning in calendar year 2024.
+  
+  *We are already getting concerned inquiries in response to CVE scans recommending upgrading to spring-framework 6. We look forward to your support of this activity*.
+
+  In order to stay on a supported version of spring-framework we need to migrate to spring-framework 6.
+
+* **Apache Wicket 10**
+
+  Apache Wicket user-interface framework is used for the GeoServer Admin console screens.
+  
+  *Brad Hards has started this activity by going to the intermediate goal of Wicket 9, and will require a fleet of testers to perform A/B testing of each screen. This is an impressive undertaking, in 2016 we did an entire round of fundraising to assemble a [team sprint]({% post_url 2016-01-25-geoserver-code-sprint-success %}) when updating from Apache Wicket 1.4. to Wicket 7.x*
+  
+  The use of spring-framework 6 and above requires the use of Apache Wicket 10.
```

Comment:
This is possibly misleading. We need Java 17 for spring 6, and Wicket <= 9 requires Java <=11. However nothing in spring requires any version of wicket, AFAICT.

Reply:
How do we fix it? Just mention the Java 17 requirement, and move this further down the page?

```
The use java 17 requires the use of Apache Wicket 10.
```

Reply:
Hum... actually I believe that current Wicket works fine with Java 17... but we definitely need Wicket 10 while upgrading to Spring 6, because Spring 6 is built on top of Jakarta EE, and Wicket 10 is the first "release" to support Jakarta EE as well. I'm saying "release" because Wicket 10 is at "milestone 2" stage, not sure if there is an ETA for it. Don't think it's going to be a problem anyways, so much work to do before we can consider an upgrade to Jakarta EE anyways.

Reply:
Probably easiest that way - its a Java 17 driven change.

Reply:
Addressed with:

> The use of Jakarta Enterprise Edition requires the use of Apache Wicket 10.